### PR TITLE
Test primary key type

### PR DESCRIPTION
**Pull request by loicgasser**
FAILED (errors=0, failures=71)

On primary key types. You can use `jenkins test` to trigger a new build, once the database or models have been fixed.

### CODE REVIEW

**Comment by rfrancel** [comment]
jenkins test

**Comment by loicgasser** [comment]
jenkins test

File: chsdi/models/vector/stopo.py [hunk 1]
```diff
@@ -1013,7 +1013,6 @@ class GridstandSwissimage(Base, Vector):
     __template__ = 'templates/htmlpopup/images_metadata.mako'
     __bodId__ = 'ch.swisstopo.images-swissimage.metadata'
     __label__ = 'lk25_name'
-    id = Column('tilenumber', Text, primary_key=True)
```
Comment (loicgasser) [inline chsdi/models/vector/stopo.py]: ???

File: chsdi/models/vector/bafu.py [hunk 2]
```diff
@@ -2018,7 +2018,7 @@ class sturm_boeenspitzen_50(Base, Vector):
     __bodId__ = 'ch.bafu.sturm-boeenspitzen_50'
     __template__ = 'templates/htmlpopup/sturm_boeenspitzen.mako'
     __label__ = 'id'
-    id = Column('oid', Integer, primary_key=True)
+    id = Column('bgdi_id', BigInteger, primary_key=True)
```
Comment (loicgasser) [inline chsdi/models/vector/bafu.py]: Why did you change the primary key?

**Comment by loicgasser** [comment]
Everything has been corrected in DB or in code. @rfrancel could you add the list of all the tables that have been modified please?

Would be great and then we can merge. We'll handle the numeric stuff in the second step.

**Comment by rfrancel** [comment]
jenkins test

**Comment by rfrancel** [comment]
jenkins test

**Comment by rfrancel** [comment]
Ok I think there are no more errors concerning different primary key types between DB and models.
For ch.bfe.sachplan-uebertragungsleitungen_anhoerung and ch.bav.sachplan-infrastruktur-schifffahrt_anhoerung, I commented the models, since the tables are empty.
I modified the bod and put tooltip = false for both layers and searchable = false for ch.bav.sachplan-infrastruktur-schifffahrt_anhoerung, since tests were failing because of the configuration. Probably need further discussion with you @AFoletti to know how to deal with these layers.
One more jenkins test to see if everything is ok now..

**Comment by rfrancel** [comment]
And here the list of the tables whose primary key was modified (and associated views):
following this structure:
*********
DB name
**********
- technical layer name
- shema.table (old type -> new type)

***********
*  STOPO  *
***********

- ch.swisstopo.swissboundaries3d-gemeinde-flaeche.fill

tlm.prodas_spatialseltype_gemeinde (id double -> integer)

NOTICE:  drop cascades to 5 other objects
DETAIL:  drop cascades to view tlm.swissbuildings_gemeinden_toposhop
drop cascades to view tlm.swissboundaries_gemeinden_uebersetzt
drop cascades to view tlm.swissboundaries_gemeinden_toposhop
drop cascades to view tlm.swissboundaries_gemeinden


- ch.swisstopo.geologie-geophysik-deklination

geol.geophysik_deklination (gid long -> integer)


- ch.swisstopo.vec25-strassennetz

public.v25_str_25_l_tooltip (objectid double -> integer)

NOTICE:  drop cascades to 3 other objects
DETAIL:  drop cascades to view v25_hptstr_25_l
drop cascades to view v25_str_25_l_tooltip
drop cascades to view v25_wanderweg

- ch.swisstopo.swissboundaries3d-kanton-flaeche.fill

tlm.prodas_spatialseltype_kanton (id double -> integer)

NOTICE:  drop cascades to 5 other objects
DETAIL:  drop cascades to view tlm.swissboundaries_kantone
drop cascades to view tlm.swissboundaries_gemeinden_uebersetzt
drop cascades to view tlm.swissbuildings_gemeinden_toposhop
drop cascades to view tlm.swissboundaries_gemeinden
drop cascades to view tlm.swissboundaries_gemeinden_toposhop

- ch.swisstopo.swissnames3d

tlm.swissnames3d_raster (bgdi_id long -> integer)

NOTICE:  drop cascades to 14 other objects
DETAIL:  drop cascades to materialized view tlm.swissnames3d_raster_00
drop cascades to materialized view tlm.swissnames3d_raster_01
drop cascades to materialized view tlm.swissnames3d_raster_02
drop cascades to materialized view tlm.swissnames3d_raster_03
drop cascades to materialized view tlm.swissnames3d_raster_04
drop cascades to materialized view tlm.swissnames3d_raster_05
drop cascades to materialized view tlm.swissnames3d_raster_06
drop cascades to materialized view tlm.swissnames3d_raster_07
drop cascades to materialized view tlm.swissnames3d_raster_08
drop cascades to materialized view tlm.swissnames3d_raster_09
drop cascades to materialized view tlm.swissnames3d_raster_10
drop cascades to materialized view tlm.swissnames3d_raster_11
drop cascades to materialized view tlm.swissnames3d_raster_12
drop cascades to materialized view tlm.swissnames3d_raster_13


- ch.swisstopo-vd.stand-oerebkataster 

vd.view_oereb_nfgeom (gid long -> integer)


- ch.swisstopo.vec25-eisenbahnnetz

public.v25_eis_25_l (objectid long -> integer)


- ch.swisstopo.geologie-geophysik-inklination gid integer

geol.geophysik_inklination (gid long -> integer)


- ch.swisstopo.vec25-heckenbaeume objectid', Integer

public.v25_heb_25_l (objectid long -> integer)


- ch.swisstopo.vec25-gewaessernetz

public.v25_gwn_25_l (objectid long -> integer)


- ch.swisstopo.swissboundaries3d-bezirk-flaeche.fill  id integer

tlm.prodas_spatialseltype_bezirk (id double -> integer)

NOTICE:  drop cascades to view tlm.swissboundaries_bezirke


- ch.swisstopo.vec25-gebaeude objectid integer

public.v25_geb_25_a (objectid double -> integer)


- ch.swisstopo.swissbuildings3d-2.metadata

public.shop_perimeter_swissbuilding3d_2 (id float -> integer)

- ch.swisstopo.geologie-geophysik-geothermie

geol.geophysik_geothermie (gid long -> integer)


- ch.swisstopo.vec25-einzelobjekte

public.v25_eob_25_l (objectid float -> integer)


- ch.swisstopo.vec25-uebrigerverkehr

public.v25_uvk_25_l (objectid float -> integer)

- ch.swisstopo.vec25-anlagen

public.v25_anl_25_a (objectid float -> integer)

- ch.swisstopo.geologie-geophysik-totalintensitaet

geol.geophysik_totalintensitaet (gid long -> integer)


*********
*  ARE  *
*********
- ch.are.agglomerationen_isolierte_staedte

siedlung_landschaft.agglomerationen_isolierte_staedte (row_id float -> integer)

- ch.are.gueteklassen_oev

oeffentlicher_verkehr.gueteklassen (id float -> integer)

- ch.are.alpenkonvention
 
siedlung_landschaft.alpenkonvention (row_id float -> integer)

- ch.are.beschaeftigtendichte

siedlung_landschaft.beschaeftigtendichte (row_id float -> integer)

- ch.are.bevoelkerungsdichte

siedlung_landschaft.bevoelkerungsdichte (row_id float -> integer)


**********
*  BAFU  *
**********

- ch.bafu.schutzgebiete-smaragd

smaragd (id bigint to integer)

- ch.bafu.swissprtr

prtr.swissprtr (prtrnr double -> Integer)

- ch.bafu.hydrologie-wassertemperaturmessstationen nr integer

hydrologie.temperaturmessnetz (nr decimal -> integer)


- ch.bafu.sturm-boeenspitzen_50

diverse.data_boeenspitzen (oid -> bgdi_id BigInteger in model)


- ch.bafu.gefahren-baugrundklassen

gefahren.baugrundklassen (_count long -> Integer)


- ch.bafu.sturm-boeenspitzen_100 oid integer

diverse.data_boeenspitzen (oid -> bgdi_id BigInteger in model)

- ch.bafu.flora-weltensutter_atlas

flora.ws (gid long -> integer)

- ch.bafu.gefahren-spektral

gefahren.baugrundkl_spectral (_count float -> integer)



**********
*  UVEK  *
**********

- ch.astra.strassenverkehrszaehlung_messstellen-uebergeordnet

astra.verkehrszaehlstellen nr (long -> integer)

NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to view astra.verkehr_ueber
drop cascades to view astra.verkehr_reg_loc

- ch.astra.ausnahmetransportrouten

astra.ausnahmetransportrouten (id long -> integer)


- ch.bakom.anbieter-eigenes_festnetz

bakom.nga_anbieter (cellid long or bigint -> integer)

NOTICE:  drop cascades to 18 other objects
DETAIL:  drop cascades to view bakom.nga_u50
drop cascades to view bakom.nga_u20
drop cascades to view bakom.nga_u2
drop cascades to view bakom.nga_u100
drop cascades to view bakom.nga_u10
drop cascades to view bakom.nga_u01
drop cascades to view bakom.nga_tv
drop cascades to view bakom.nga_hdtv
drop cascades to view bakom.nga_fiber
drop cascades to view bakom.nga_d50
drop cascades to view bakom.nga_d20
drop cascades to view bakom.nga_d2
drop cascades to view bakom.nga_d100
drop cascades to view bakom.nga_d10
drop cascades to view bakom.nga_d1
drop cascades to view bakom.nga_copper
drop cascades to view bakom.nga_coax
drop cascades to view bakom.nga_anbieter

- ch.bfe.sachplan-geologie-tiefenlager-thematische-darstellung

bfe.geologische_tiefenlager_fac (stabil_id text -> integer)
bfe.geologische_tiefenlager (stabil_id text -> integer)
bfe.geologische_tiefenlager_raster (stabil_id text -> integer)

- ch.bfe.stauanlagen-bundesaufsicht

bfe.stauanlagen_bundesaufsicht (dam_stabil_id long -> integer)

- ch.bfe.statistik-wasserkraftanlagen

bfe.statistik_wasserkraftanlagen_powerplant (wastanumber long -> integer)

- ch.bfe.solarenergie-eignung-daecher

bfe.solarenergie_daecher (df_uid long -> integer)

NOTICE:  drop cascades to 3 other objects
DETAIL:  drop cascades to view bfe.view_solarenergie_daecher_gs
drop cascades to view bfe.view_solarenergie_daecher_gs_ltclm
drop cascades to view bfe.view_solarenergie_daecher_gs_replaced

************
*  DRITTE  *
************

- ch.ensi.zonenplan-notfallschutz-kernanlagen

ensi.zonenplan_kernanlagen (nr long -> integer)

**Comment by loicgasser** [comment]
Yes great work @rfrancel.  I ll test a bit more tomorrow and solve this empty table issue with ambrogio. Thks @ltclm for the update.
